File: 프로그래머스/lv1/12947. 하샤드 수/하샤드 수.cpp
#include <string>
#include <vector>

using namespace std;

bool solution(int x)
{
    bool answer = true;
    string s = to_string(x);
    int sum = 0;
    for (int i = 0; i < s.length(); i++)
        sum += s[i] - 48;

    if (x % sum != 0)
        answer = false;

    return answer;
}

File: 프로그래머스/1/12947. 하샤드 수/하샤드 수.java
class Solution {
    public boolean solution(int x) {
        int y = x;
        int sum = 0;
        
        while (y >= 1) {
            int rem = y % 10;
            y /= 10;
            sum += rem;
        }
        
        return x % sum == 0 ? true : false;
    }
}